exporter/prometheusexporter/collector.go

```diff
@@ -40,6 +43,7 @@ func newCollector(config *Config, logger *zap.Logger) *collector {
 		sendTimestamps:    config.SendTimestamps,
 		constLabels:       config.ConstLabels,
 		addMetricSuffixes: config.AddMetricSuffixes,
+		metricFamilies:    make(map[string]*dto.MetricFamily),
```

Comment:
I think we will need to "expire" these to ensure we don't leak memory, similar to the accumulator above.

Reply:
Updated to clean up expired metrics - should this also be a sync.Map?

Reply:
Yes, it does need to be a sync.Map